### PR TITLE
Avoiding textplain passwords

### PR DESCRIPTION
**Pull request by v-fearam**
Reference:
https://learn.microsoft.com/azure/azure-resource-manager/bicep/scenarios-secrets

Outputs are logged to the deployment history, and anyone with access to the deployment can view the values of a deployment's outputs.

When you need to provide secrets to your Bicep deployments as parameters, [use the @secure() decorator](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters#secure-parameters). When you mark a parameter as secure, Azure Resource Manager avoids logging the value or displaying it in the Azure portal, Azure CLI, or Azure PowerShell.

Extra:
   Update API, removing some bicep warning

### CODE REVIEW

File: Bicep Deployment/_modules/cosmosdb.bicep [hunk 1]
```diff
@@ -21,9 +23,9 @@ resource cosmosDB 'Microsoft.DocumentDB/databaseAccounts@2021-06-15' = {
     }
     locations: [
       {
-        locationName: resourceGroup().location
+        locationName: rgLocation
         failoverPriority: 0
-        isZoneRedundant: true
+        isZoneRedundant: false
```
Comment (ckittel) [inline Bicep Deployment/_modules/cosmosdb.bicep:28]: Why this change?

File: Bicep Deployment/_modules/functions.bicep [hunk 2]
```diff
@@ -1,38 +1,45 @@
...
 param appGWName string
 param appInsightInstrumentationKey string
 param acrServerName string
 param acrLoginName string
-param acrPassword string
 param networkProfileName string
 param appgw_bepool_name string
 param appgw_listener_name string
 param appgw_httpsetting_name string
+param location string =  resourceGroup().location
```
Comment (ckittel) [inline Bicep Deployment/_modules/functions.bicep:13]: In some you named it `rgLocation` and here `location` please pick just one (I suggest just `location`).

File: Bicep Deployment/main.bicep [hunk 3]
```diff
@@ -69,17 +76,20 @@ module functions '_modules/functions.bicep' = {
     appInsightInstrumentationKey: monitoring.outputs.instrumentationKey
     acrServerName: acr.outputs.acrServerName
     acrLoginName: acr.outputs.acrLoginName
-    acrPassword: acr.outputs.acrPassword
```
Comment (ckittel) [inline Bicep Deployment/main.bicep:72]: Can you please remove this from the outputs then as well.  I don't think it's used for anything else (but you'd need to check)

**Comment by ckittel** [comment]
BTW, I just reached out to @guangying94 yesterday for him to do some maintenance. So, there should be some more work being done in a new PR later to do a full update pass.  that has NOTHING to do with this PR, just FYI.

File: Bicep Deployment/main.bicep (hunk 3 above)
Comment (v-fearam) [reply Bicep Deployment/main.bicep:72]: It was already removed, 
Bicep Deployment/_modules/containers.bicep line 35

File: Bicep Deployment/_modules/cosmosdb.bicep (hunk 1 above)
Comment (v-fearam) [reply Bicep Deployment/_modules/cosmosdb.bicep:28]: Good catch, I will rollback. 
I have an error (something like, Sorry, we are currently experiencing high demand in West US 2 region for the zonal redundant (Availability Zones) accounts, and cannot fulfill your request at this time Thu, 22 Feb 2024 14:37:32 GMT.) on our subscription using true, I changed for testing.

**Review by ckittel — APPROVED** [review]
🚀
